clinica/iotools/converter_utils.py

```diff
@@ -1,13 +1,9 @@
 def sort_session_list(session_list):
-    session_idx = [int(session[5:]) for session in session_list]
-    session_idx.sort()
     session_id_list = []
+    session_idx = [((session[5:]), int(session[5:])) for session in session_list]
```

Comment:
I would avoid the magic 5 here since it's not obvious what is is at first glance.
I would add type hints, and a docstring. Also, I think you can rewrite the function in a more concise way.
Something like this maybe:

```python
def sort_session_list(session_list: list[str]) -> list[str]:
    """Sorts the list of session IDs provided based on their session number.

    Parameters
    ------------
    session_list : list[str]
        List of session IDs to sort.

    Returns
    --------
    list[str] : 
        Sorted list of session IDs.
    """
    prefix_length = len("ses-M")
    session_idx = [((session[prefix_length:]), int(session[prefix_length:])) for session in session_list]
    session_idx.sort(key=lambda x: x[1])
    return [f"ses-M{session[0]}" for session in session_idx]
```

Finally it'd be great to add some tests for this function.

Reply:
Yes that makes sense !